tests/measurements/test_classical_shadow.py

```diff
@@ -116,6 +116,117 @@ def circuit():
 wires_list = [1, 3]
 
 
+class TestProcessState:
+    """Unit tests for process_state_with_shots for the classical_shadow
+    and shadow_expval measurements"""
+
+    def test_shape_and_dtype(self):
+        """Test that the shape and dtype of the measurement is correct"""
+        mp = qml.classical_shadow(wires=[0, 1])
+        res = mp.process_state_with_shots(np.ones((2, 2)) / 2, qml.wires.Wires([0, 1]), shots=100)
+
+        assert res.shape == (2, 100, 2)
+        assert res.dtype == np.int8
+
+        # test that the bits are either 0 and 1
+        assert np.all(np.logical_or(res[0] == 0, res[0] == 1))
+
+        # test that the recipes are either 0, 1, or 2 (X, Y, or Z)
+        assert np.all(np.logical_or(np.logical_or(res[1] == 0, res[1] == 1), res[1] == 2))
+
+    def test_wire_order(self):
+        """Test that the wire order is respected"""
+        state = np.array([[1, 1], [0, 0]]) / np.sqrt(2)
+
+        mp = qml.classical_shadow(wires=[0, 1])
+        res = mp.process_state_with_shots(state, qml.wires.Wires([0, 1]), shots=1000)
+
+        assert res.shape == (2, 1000, 2)
+        assert res.dtype == np.int8
+
+        # test that the first qubit samples are all 0s when the recipe is Z
+        assert np.all(res[0][res[1, ..., 0] == 2][:, 0] == 0)
+
+        # test that the second qubit samples contain 1s when the recipe is Z
+        assert np.any(res[0][res[1, ..., 1] == 2][:, 1] == 1)
+
+        res = mp.process_state_with_shots(state, qml.wires.Wires([1, 0]), shots=1000)
```

Comment:
I think it could also be a good idea to test `process_state_with_shots` on a larger state than the measurement process. But that was just a thought I had. Feel free to ignore this.

Reply:
That sounds like a good thing to test 👍